heuristic/classes/Route.py

```diff
@@ -51,3 +52,22 @@ def _plan_cost(self, problem: Problem) -> float:
             cost += Stacks.cost(customer, problem, before, after)
 
         return cost
+
+    def remove_customer(self, customer: int):
+        """
+        Removes customer from route and items with the customer as destination
+        or origin from the loading plan.
+        """
+        idx = self.customers.index(customer)
+        delivery_item = Item(Problem.demands[customer], DEPOT, customer)
+        pickup_item = Item(Problem.demands[customer], customer, DEPOT)
+
+        for stacks in self.plan[:(idx - 1)]:
```

Comment:
Be _very_ careful about the offsets here - I do not know them by heart, and you'll need to test this to find out if this is the correct cut-off point. I think it's too early, but again, I could be wrong.

Reply:
In the graphs, at idx there was only a pickup present, which is why I thought this would be the cut-off point.

Reply:
Alright then!

Reply:
The first loading plan is always just after leaving the depot, so this should run up to `idx`, not `idx - 1`. Similarly for the pickup item, which should be in the tour from `idx + 1` onwards.

Reply:
OK I need to think about this a little more.

Reply:
Yeah it's up to `idx + 1` for deliveries, and from `idx + 1` onwards for pickups. See e.g. [this build](https://travis-ci.com/N-Wouda/OR-Analysis/jobs/288094552) where I have updated the validator to consider this.